scripts/audio/AudioEngine.cs

```diff
@@ -0,0 +1,285 @@
+        {
+            _musicPlayer1.Stream = audioStream;
+            _musicPlayer1.VolumeDb = 0;
+            _musicPlayer1.Play();
+        }
+
+        else if (!_musicPlayer1.Playing) // if channel 1 is not playing, but 2 is, cross fade to channel 1
+        {
+            _musicPlayer1.Stream = audioStream;
+            _ = FadeInTrack(_musicPlayer1); // assigning Task to discard to make rider happy, as I'm not awaiting async
+            if (_musicPlayer2.Playing)
+            {
+                _ = FadeOutTrack(_musicPlayer2);
+            }
+        }
+        else if (!_musicPlayer2.Playing) // if channel 1 is playing, but not 2, cross fade to channel 2
+        {
+            _musicPlayer2.Stream = audioStream;
+            _ = FadeInTrack(_musicPlayer2);
+            _ = FadeOutTrack(_musicPlayer1);
+        }
+        else // if both channels are playing, just stop all music and play on channel 1
+        {
+            _musicPlayer1.Stop();
+            _musicPlayer2.Stop();
+            _musicPlayer1.Stream = audioStream;
+            _musicPlayer1.Play();
+        }
+    }
+
+    private async Task FadeOutTrack(AudioStreamPlayer player) // -80 is the lowest volume, 0 is the highest
+    {
+        float vol = player.VolumeDb;
+        while (vol > -80)
+        {
+            vol -= 0.1f;
+            player.VolumeDb = vol;
+
+            await Task.Delay(30);
+            if (vol <= -80)
+            {
+                player.Stop();
+            }
+        }
+    }
+
+    private async Task FadeInTrack(AudioStreamPlayer player)
+    {
+        float vol = player.VolumeDb;
+        vol -= 80f;
+        player.VolumeDb = vol;
+        player.Play();
+
+        while (vol < 0)
+        {
+            vol += 1f;
+            player.VolumeDb = vol;
+            await Task.Delay(30);
+            if (vol >= 0)
+            {
+                break;
+            }
+        }
+    }
+
+    public async Task FadeAllTracks()
+    {
+        await FadeOutTrack(_musicPlayer1);
+        await FadeOutTrack(_musicPlayer2);
+    }
+
+    public void StopAllTracks()
+    {
+        if (_musicPlayer1.Playing)
+        {
+            _musicPlayer1.Stop();
+        }
+
+        if (_musicPlayer2.Playing)
+        {
+            _musicPlayer2.Stop();
+        }
+    }
+
+    // Sound Effect Methods //
+    public void PlaySoundFx(string soundFxFileName)
+    {
+        AudioStream audioStream = GetAudioStream(soundFxFileName, AudioType.SoundFx);
+
+        if (!_soundFxPlayer1.Playing) // if nothing is playing, play on channel 1
+        {
+            _soundFxPlayer1.Stream = audioStream;
+            _soundFxPlayer1.Play();
+        }
+        else if (!_soundFxPlayer2.Playing) // if channel 1 playing, play on 2
+        {
+            _soundFxPlayer2.Stream = audioStream;
+            _soundFxPlayer2.Play();
+        }
+        //otherwise, do nothing, 2 sound effects at once is enough
+
+        else
```

Comment:
else should be next to if bracket, it's clearer what it's affected by

(If I didn't have other comments here I wouldn't block like this.

Reply:
good catch, This happened because of where I put the comment. code reformat added a space because of it. Moved the comment next to else instead